library/include/borealis/core/touch/gesture_recognizer.hpp

```diff
@@ -0,0 +1,63 @@
+/*
+    Copyright 2021 XITRIX
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+#pragma once
+
+#include <functional>
+
+#include <borealis/core/input.hpp>
+
+namespace brls
+{
+
+class View;
+
+// Represents current gesture state
+enum class GestureState
+{
+    INTERRUPTED, // Gesture has been interupted, no callbacks will come
+    UNSURE, // Gesture started recognition and not sure if it should interupt other recognizers
```

Comment:
I don't find the name UNSURE alone very self explanatory, how about something like START_UNSURE_INTERRUPT?

Reply:
I think it's quite self explanatory, i.e. I just hold my finger on screen and recognizer is UNSURE what I want to do. I mean it can't tell does it should fail or start tracking that gesture, so it is UNSURE. It is not about, is it sure that it should stop other recognizers.